### ISSUE TITLE
ERROR: Datadog API key must consist of 32 hexadecimal digits

### ISSUE THREAD

**Issue by kirkiris** [mongodb/terraform-provider-mongodbatlas#1001]
### Terraform CLI and Terraform MongoDB Atlas Provider Version

```hcl
# Copy-paste your version info here
Terraform v1.2.4
on linux_amd64

mongodbatlas = {
  source = "mongodb/mongodbatlas"
  version = "1.6.1"
```

### Terraform Configuration File 

```hcl
# Copy-paste your configuration info here
provider "mongodbatlas" {
  public_key  = var.mongodbatlas_public_key
  private_key = var.mongodbatlas_private_key
}
```

### Steps to Reproduce

1. Create a mongodbatlas_alert_configuration with `notification.type_name = "DATADOG"`
2. `terraform init`
3. `terraform plan`
4. `terraform apply`

### Expected Behavior

`terraform apply` should create/update the corresponding resources
### Actual Behavior

`terraform apply` fails with the following error:
> Error: error updating Alert Configuration information: PUT https://cloud.mongodb.com/api/atlas/v1.0/groups/xxx/alertConfigs/yyy: 400 (request "INVALID_DATADOG_API_KEY") Datadog API key must consist of 32 hexadecimal digits.

### Debug Output 

```
---[ REQUEST ]---------------------------------------
PUT [REDACTED] HTTP/1.1
Host: cloud.mongodb.com
User-Agent: terraform-provider-mongodbatlas/1.6.1 go-mongodbatlas/0.18.0 (linux;amd64)
Content-Length: 356
Accept: application/json
Content-Type: application/json
Accept-Encoding: gzip

{
 "id": "[REDACTED]",
 "eventTypeName": "HOST_DOWN",
 "enabled": true,
 "notifications": [
  {
   "datadogApiKey": "[REDACTED]",
   "delayMin": 0,
   "emailEnabled": false,
   "intervalMin": 720,
   "smsEnabled": false,
   "typeName": "DATADOG"
  },
  {
   "delayMin": 0,
   "emailEnabled": true,
   "intervalMin": 720,
   "smsEnabled": false,
   "typeName": "GROUP",
   "roles": [
    "GROUP_OWNER"
   ]
  }
 ]
}
```

```
---[ RESPONSE ]--------------------------------------
HTTP/2.0 400 Bad Request
Content-Length: 156
Content-Type: application/json
Date: Tue, 10 Jan 2023 12:35:26 GMT
Referrer-Policy: strict-origin-when-cross-origin
Server: envoy
Strict-Transport-Security: max-age=31536000; includeSubdomains;
X-Content-Type-Options: nosniff
X-Envoy-Upstream-Service-Time: 27
X-Frame-Options: DENY
X-Mongodb-Service-Version: gitHash=5d8e1068634c830d6984e83071ef22d155814c26; versionString=v20230105
X-Permitted-Cross-Domain-Policies: none

{
 "detail": "Datadog API key must consist of 32 hexadecimal digits.",
 "error": 400,
 "errorCode": "INVALID_DATADOG_API_KEY",
 "parameters": [],
 "reason": "Bad Request"
}
```
### Additional Context

I've tried the same request with `curl` and it succeeded.

**Comment by martinstibbe** [comment]
@kirkiris This may be due to an issue with API returning obfuscated value for API in state file
we are fixing this in v1.8.0 release of provider fixed in Engineering ticket [INTMDB-519](https://jira.mongodb.org/browse/INTMDB-519)
Workaround is to set ignore changes on notifications[1].datadogApiKey until fix is released

**Comment by Zuhairahmed** [comment]
@kirkiris as an update, this issue has been picked up in current sprint cycle and you can expect fix to be included as part of v1.8.0 release next week. will let you know when latest release has been published to the Terraform Registry.

**Comment by Zuhairahmed** [comment]
@kirkiris we just released v1.8.0, feel free to give it try! issue should have been resolved.

**Comment by Zuhairahmed** [comment]
Closing as appears to be duplicate to #1049, feel free to re-open if you need anything else
